docs/SelfDiagnostics.MD

```diff
@@ -0,0 +1,65 @@
+# [Advanced] How to enable self diagnostics for ApplicationInsights.Kubernetes
+
+## Enable self-diagnostics
+
+We understand ApplicationInsights.Kubernetes is far from perfect. Enabling self-diagnostics will help find [issues](https://github.com/microsoft/ApplicationInsights-Kubernetes/issues) easier. Please feel free to open issues and we will be glad to look into them.
+
+Application Insights for Kubernetes is instrumented with DiagnosticsSource. To see the logs, create a diagnostic source listener that listens to `Microsoft.ApplicationInsights.Kubernetes.Debugging.Inspect.Observer`. The observer(DiagnosticSource) will emits logs in different levels.
+
+To make diagnostic easier, a default listener that outputs logs into the console is provided and and be enabled like it below before `AddApplicationInsightsKubernetesEnricher` is called:
+
+```csharp
+var listener = InspectListener.Instance;
+listener.SetMinimumLevel(InspectLevel.Trace);
+Inspect.Instance.Observer.SubscribeWithAdapter(listener);
+```
+
+When the listener above is enabled, you will see diagnostic logs like this:
+
+```shell
+...
+[Debug] Application Insights Kubernetes injected the service successfully.
```

Comment:
Consider

- adding timestamp to logs
- somehow indicating the log comes from the AIKObserver (as opposed to all the other noise written to std output stream).

Reply:
Will consider the implementation separately.